scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -141,8 +144,8 @@ private IteratingResult nextValueAt(
           }
         }
         cur = chooseRandomX(min, max);
-        //if min == max, another call to random will have no effect and thus we should exit
-      } while(i < maxIteration && !min.isEqualTo(max));
+        //if all timepoints have been visited or min == max, another call to random will have no effect and thus we should exit
+      } while(i < maxIters);
```

Comment:
why removing the second !min.isEqualTo(max) ?

Reply:
if min == max, `numTimepoints` will be -1, so `maxIters` will be -1

Reply:
can you then remove the "or min == max" part of the comment ?

Reply:
removed